[c-wenlong] iP

c-wenlong opened the pull request:
# Kaipybara Bot

```
          /'  _/                                /'
        /' _/~                                /'
     ,/'_/~  ____     O   ____              /'__     ____      ____     ____
    /\/~   /'    )  /'  /'    )--/'    /  /'    )  /'    )   )'    )--/'    )
  /'  \  /'    /' /'  /'    /' /'    /' /'    /' /'    /'  /'       /'    /'
/'     \(___,/(__(__/(___,/'  (___,/(__(___,/(__(___,/(__/'        (___,/(__
                  /'             /'
                /'       /     /'
              /'        (___,/'
```
> Your Best Bot and Friend 

About Me
-
I am a bot that helps you organise your tasks and boost your productivity so that you can focus on the more important things in life, such as watching your favourite TV show. I can:
- Able to read your text inputs.
- Execute commands to help you organise your life.
- Keep it saved locally once you leave the program.

All you need to do is to [download](https://github.com/c-wenlong/ip) me! :shit: And your life will become so much __*better*__.
1. No longer worried about forgetting deadlines.
2. No longer missing your dates with your potential partners.
3. No longer missing your meds.

Features
-
- [X] Add, Delete, Mark & Unmark Tasks.
- [X] View all Tasks.
- [X] Search for Tasks.

If you are a Java programmer
-
You can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Ui ui = new Ui();
        System.out.println(ui.getStartMessage()); // opening statement
        TaskManager.loadLocalSavedTasks(); // checking if there are local files to load
        KBot.simulate(); // simulate kaipybara chatbot
        System.out.println(ui.getEndMessage()); // closing statement
        System.out.println(ui.getFlower());
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

c-wenlong committed:
Level 0. Rename, Greet, Exit

c-wenlong committed:
Level 1. Echo

c-wenlong committed:
Level 2. Add, List

c-wenlong committed:
Level 3. Mark as Done

c-wenlong committed:
Level 4. ToDos, Events, Deadlines

c-wenlong committed:
A-TextUiTesting

c-wenlong committed:
Level 5. Handle Errors

c-wenlong committed:
Level 6. Delete

c-wenlong committed:
Save the tasks in the hard disk automatically whenever the task list changes. Load the data from the hard disk when the chatbot starts up.

c-wenlong committed:
Save the tasks in the hard disk automatically whenever the task list changes. Load the data from the hard disk when the chatbot starts up.

c-wenlong committed:
Merge branch 'branch-Level-7'

Save the tasks in the hard disk automatically whenever the task list changes. Load the data from the hard disk when the chatbot starts up.

c-wenlong committed:
Teach the chatbot how to understand dates and times

c-wenlong committed:
Merge branch 'add-gradle-support'

c-wenlong committed:
Add gradle configurations

c-wenlong committed:
Add tests for tasks

c-wenlong committed:
Add jar file

c-wenlong committed:
Give users a way to find a task by searching for a keyword

c-wenlong committed:
Merge branch 'Level-9'

c-wenlong committed:
Change help function

c-wenlong committed:
Add more OOP

c-wenlong committed:
Merge branch 'A-MoreOOP'
Add more OOP

c-wenlong committed:
Add more packaging

c-wenlong committed:
Merge branch 'A-Packages'

c-wenlong committed:
Add Java documentation

c-wenlong committed:
Merge branch 'branch-A-JavaDoc'

c-wenlong committed:
Remove redundant code

c-wenlong committed:
Merge branch 'branch-A-JavaDoc'
Merge with branch-A-JavaDoc

c-wenlong committed:
Merge branch 'master' into branch-A-CodingStandard
Merge with master

c-wenlong committed:
Make code fit java coding standard

c-wenlong committed:
Merge branch 'branch-A-CodingStandard'
Merge with branch-A-CodingStanard to make code more standardised

c-wenlong committed:
Create GUI for chatbot

c-wenlong committed:
Merge branch 'branch-Level-10'
Create GUI for chatbot

TopKec commented on src/main/java/duke/KBot/Parser.java:
```diff
@@ -0,0 +1,77 @@
+                    throw new InvalidInputException("Error: " + "Invalid parameters for " + instruction);
+                default: // incorrect ins and no param handled here
+                    throw new InvalidCommandException("Invalid command: " + instruction
+                            + "\nPlease input the correct commands. Input help to see list of commands.");
```
I do like how instructions for recognized comments is handled for, though it might be worth putting "help" in inverted commas? Something like
```
...Input "help" to see a list of commands.
```
Might be pedantic, but could potentially improve user experience?

TopKec commented on src/main/java/duke/KBot/TaskManager.java:
```diff
@@ -0,0 +1,48 @@
+/**
+ * Manages all communications between the bot simulation and the database of
+ * Tasks. Inspired by ChatGPT, but I simply just created this class and
+ * transferred all my original code over from KBot.java
```
Love the credit to ChatGPT, haha!

TopKec commented on src/main/java/duke/actions/AddTask.java:
```diff
@@ -0,0 +1,104 @@
+            case "deadline":
+                String[] deadlineParameter = parameter.split(" /by ", 2);
+                if (deadlineParameter.length == 1) { // if there is command but no input
+                    throw new InvalidInputException("Invalid parameters for " + instruction);
```
Might it be worth noting more precisely how the parameters are invalid? So the user knows how exactly they went wrong in the input. Alternatively, a comment to just say `Type "help" if you're unsure` could improve user experience?

TopKec commented on src/main/java/duke/KBot/TaskFileManager.java:
```diff
@@ -0,0 +1,126 @@
+     */
+    public static Task loadTask(String instruction, String parameter) {
+        Task t = null;
+        switch (instruction) {
```
(Generally for switch statements)
I think that the indentation on the "case" conditions are off slightly? They seem to be 4 spaces too far.

TopKec reviewed (commented):
Excellent coding style in general, couldn't find very many issues with the coding style and such. Stuff like names, line length and many others were all adhered to very well!

naozee commented on .vscode/settings.json:
```diff
@@ -0,0 +1,3 @@
+{
+    "java.configuration.updateBuildConfiguration": "interactive"
```
Not sure if this file is relevant. Might want to gitignore the directory?

naozee commented on src/main/java/duke/KBot/TaskFileManager.java:
```diff
@@ -0,0 +1,126 @@
+    private static final String FILEPATH = "src/main/java/duke/memory/output.txt";
+
+    /** Storage format for date. */
+    private static final DateTimeFormatter STORAGEFORMAT = DateTimeFormatter.ofPattern("d-M-yy");
```
Might want to use use FILE_PATH and STORAGE_FORMATTER instead

naozee commented on src/main/java/duke/KBot/DialogBox.java:
```diff
@@ -0,0 +1,69 @@
+     * Flips the dialog box such that the ImageView is on the left and text on the
+     * right.
+     */
+    private void flip() {
```
May want to use flipDialogBox instead?

naozee reviewed (commented):
Naming is pretty good :)

c-wenlong committed:
Add build files from Gradle

By adding the build files with a JAR file, anyone can run the program
now with ease, through the JAR file.

You can find it in /build/libs/*.

c-wenlong committed:
Add some assertions to better control flow

Assertions are important because it helps to determine where errors and
bugs may be in the flow of the program. I have added it at key points to
check that userInput is not null.

c-wenlong committed:
Improve on code quality

c-wenlong committed:
Merge pull request #1 from c-wenlong/branch-A-Assertions

Add some assertions to better control flow

c-wenlong committed:
Fix merge conflicts

c-wenlong committed:
Merge pull request #2 from c-wenlong/branch-A-CodeQuality

Improve on code quality

c-wenlong committed:
Delete un-needed file

c-wenlong committed:
Add tags as an input option

c-wenlong committed:
Tidy up JavaDoc and fix bugs in C-Tagging

c-wenlong committed:
Stop tracking .class files

c-wenlong committed:
Remove previously uploaded files that are not supposed to be tracked

c-wenlong committed:
Fix tagging bug

c-wenlong committed:
Improve JavaDoc and code quality

c-wenlong committed:
Fix tagging bug

c-wenlong committed:
Create README for github pages

c-wenlong committed:
Fix gradlew build path

c-wenlong committed:
Trivial change to README doc

c-wenlong committed:
Fix relative path bug

c-wenlong committed:
Add Ui.png and table of contents to page

c-wenlong committed:
Change file names

c-wenlong committed:
Fix file paths

c-wenlong committed:
Fix program to run on Intel

c-wenlong committed:
Update UG and test

c-wenlong committed:
Reduce long methods

c-wenlong committed:
Add package parser

c-wenlong committed:
Reduce method length in AddTask

c-wenlong committed:
Refractor code
